### ISSUE TITLE
Progress bar prints out progress bar twice

### ISSUE THREAD

**Issue by terrytangyuan** [rstudio/tfestimators#82]
This happens when running the custom model test https://github.com/rstudio/tfestimators/blob/master/tests/testthat/test-tf-custom-models.R#L53
```
> classifier <-
+     estimator(model_fn = simple_custom_model_fn, model_dir = tmp_dir) %>%
+     train(input_fn = constructed_input_fn, steps = 2L)
2/2 [==================================================] - ETA:  0s - loss: 0.87
2/2 [==================================================] - ETA:  0s - loss: 0.87
```

**Comment by terrytangyuan** [comment]
Fixed via https://github.com/rstudio/tfestimators/commit/3d1f314694701f179be2b19da3d24271c156285d.

**Comment by terrytangyuan** [comment]
The commit I made was not correct. It only fixed situation where steps = 2. However, when `steps = 10`, we'll see the following where the progress bar is not completed:
```
> classifier <-
+     estimator(model_fn = simple_custom_model_fn, model_dir = tmp_dir) %>%
+     train(input_fn = constructed_input_fn, steps = 10L)
8/10 [=======================================..........] - ETA:  0s - loss: 0.62
```

Seems like the duplicate progress bar only happens when `steps <= 8`, which is when input is exhausted.

**Comment by terrytangyuan** [comment]
Just fixed this in 110e7a5. @kevinushey Could you double check to see if I fixed in the correct place?
